extensions/objects/templates/literally-canvas-diagram-editor.directive.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for literally canvas diagram editor.
+ */
+
+const LC = require('literallycanvas');
+require('literallycanvas/lib/css/literallycanvas.css');
+require('services/literally-canvas-helper.service.ts');
+
+angular.module('oppia').directive('literallyCanvasDiagramEditor', [
+  '$timeout', 'LiterallyCanvasHelperService',
+  function($timeout, LiterallyCanvasHelperService) {
+    return {
+      restrict: 'E',
```

Comment:
I think we're converting all directives to components and getting rid of `restrict: 'E''`. @marianazangrossi can advise regarding this.

Reply:
This comment has not been addressed @rohitkatlaa.

Reply:
I have changed it to a component. Done.